app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -150,6 +154,11 @@ protected void onCreate(Bundle savedInstanceState) {
             public void onClick(View arg0) {
                 errorBox.setVisibility(View.GONE);
                 ViewUtil.hideVirtualKeyboard(LoginActivity.this);
+
+                if (ResourceInstallUtils.isUpdateInstallReady()) {
+                    installPendingUpdate();
+                }

```

Comment:
This is an awkward place for installing a downloaded update, because if you give incorrect creds but there is a pending update, it will still install it. I'm going to punt on this for now because the LoginActivity is so convoluted that it is hard for me to know where a good place to put this is. I would like to refactor LoginActivity, at which point I will move this.